### PR TITLE
fix(www): layout for `/guidelines/typography` + convert to theme-ui

### PR DESCRIPTION
**Pull request by abhishekjakhar**
## Description

1. Converted the `guidelines/typography` page to theme-ui
2. Fixed the responsive layout
3. Fixed the blog cards and their responsive layout

Local URL: http://localhost:8000/guidelines/typography/
Production URL: https://www.gatsbyjs.org/guidelines/typography/

##### Desktop
desktop-actual-1
desktop-actual-2
desktop-actual-3

##### Mobile
mobile-actual-1
mobile-actual-2
mobile-actual-3

##### Desktop
desktop-expected-1
desktop-expected-2
desktop-expected-3

##### Mobile
mobile-expected-1
mobile-expected-2
mobile-expected-3

### CODE REVIEW

File: www/src/components/guidelines/cards/blog.js [hunk 1]
```diff
@@ -1,21 +1,47 @@
...
+import { mediaQueries } from "gatsby-design-tokens/dist/theme-gatsbyjs-org"
 import { MdArrowForward } from "react-icons/md"
 
-import { Box, Flex, Heading, Link, Text } from "../system"
+import { Box } from "theme-ui"
+import { Flex, Heading, Link, Text } from "../system"
 import ImagePlaceholder from "../image-placeholder"
 import Avatar from "../avatar"
 
-const BlogCard = ({ ...props }) => (
```
Comment (anishagg17) [inline www/src/components/guidelines/cards/blog.js:8]: why are `props` removed  ?

Comment (abhishekjakhar) [reply www/src/components/guidelines/cards/blog.js:8]: We were passing mx={6} as props here, but I only wanted mx to be 6 above the breakpoints of sm, below that it should be 0 to fix the responsive layout. Also this is very subjective, but the thought which I had while making this change was that the blog card and starter card are very specific to `/guidelines/typography` here, and we are not using them anywhere else. I don't see any reason for passing mx={6} as props to them, so I went ahead and set mx={6} explicitly above the sm breakpoints.

**Comment by abhishekjakhar** [comment]
> Pushed a couple minor things to get this ready to merge! ✌️
> Thank you so much @abhishekjakhar, and @anishagg17 for the review help! 🙏

@fk Thank you very much.

**Comment by abhishekjakhar** [comment]
@fk What is the meaning of chore?

**Comment by fk** [comment]
@abhishekjakhar Good question! A chore is a routine task … but lemme check real quick (not a native English speaker) … yep, Google says the former, and also "a tedious but necessary task".

All the docs I can find on `chore` in the context of https://www.conventionalcommits.org/, which we are using to automatically generate `CHANGELOG`s for Gatsby and its `packages`, are a bit thin, and TBH I tried to find out a couple of times by now, without gaining much security … 😅 

In hindsight, a lot of the `chore`s I authored probably should have been `refactor`s instead — I refactored production code, but didn't change the outcome of that code. `chore` IMO is OK for these type of changes though, too, and seems to be the more common one.

Probably good to mention: When we merge PRs, we are squashing them, so all the individual commit messages of the PR end up as one commit. The PRs title becomes the title for that commit, and that one is the important one to get right. And for `www` the latter doesn't necessarily have to be a commit message following the conventional commits spec, because we do not generate a CHANGELOG for `www`.

**Comment by abhishekjakhar** [comment]
> @abhishekjakhar Good question! A chore is a routine task … but lemme check real quick (not a native English speaker) … yep, Google says the former, and also "a tedious but necessary task".
> 
> All the docs I can find on `chore` in the context of https://www.conventionalcommits.org/, which we are using to automatically generate `CHANGELOG`s for Gatsby and its `packages`, are a bit thin, and TBH I tried to find out a couple of times by now, without gaining much security … 😅
> 
> In hindsight, a lot of the `chore`s I authored probably should have been `refactor`s instead — I refactored production code, but didn't change the outcome of that code. `chore` IMO is OK for these type of changes though, too, and seems to be the more common one.
> 
> Probably good to mention: When we merge PRs, we are squashing them, so all the individual commit messages of the PR end up as one commit. The PRs title becomes the title for that commit, and that one is the important one to get right. And for `www` the latter doesn't necessarily have to be a commit message following the conventional commits spec, because we do not generate a CHANGELOG for `www`.

Very detailed answer, thank you very much.

**Pull request merged**